incubator/hnc/internal/reconcilers/object.go

```diff
@@ -570,8 +570,13 @@ func (r *ObjectReconciler) writeObject(ctx context.Context, log logr.Logger, ins
 	// The object exists if CreationTimestamp is set. This flag enables us to have only 1 API call.
 	exist := inst.GetCreationTimestamp() != v1.Time{}
 	ns := inst.GetNamespace()
+	// Get current ResourceVersion, required for smooth updates.
+	rv := inst.GetResourceVersion()
 	inst = object.Canonical(srcInst)
```

Comment:
IIUC, we should comment like below, right?
```
	// Get current ResourceVersion, required for updates for newer API objects (including custom resources).
	rv := inst.GetResourceVersion()
	// Overwrite the propagated copy with the source, then restore all essential properties of the copy.
	inst = object.Canonical(srcInst)
```

Reply:
Yup, though I'd ask for a blank line before the "Overwrite the propagated copy" - e.g. start a new "paragraph" since it's doing a different set of stuff. I know it's a minor nit but I find that paragraphs make the code a lot easier to read!